OneSignalSDK/onesignal/location/src/main/java/com/onesignal/location/internal/LocationManager.kt

```diff
@@ -22,13 +23,17 @@ internal class LocationManager(
     private val _capturer: ILocationCapturer,
     private val _locationController: ILocationController,
     private val _locationPermissionController: LocationPermissionController,
+    private val _prefs: IPreferencesService,
 ) : ILocationManager, IStartableService, ILocationPermissionChangedHandler {
-    private var _isShared: Boolean = false
+    private var _isShared: Boolean = _prefs.getBool("OneSignal", "PREFS_OS_LOCATION_SHARED", false)!!
```

Comment:
Let's use the `PREFS_OS_LOCATION_SHARED` constant you created instead of hardcoded strings:
```
_prefs.getBool(PreferenceStores.ONESIGNAL, PreferenceOneSignalKeys.PREFS_OS_LOCATION_SHARED, false)!!
```